airsonic-main/src/main/java/org/airsonic/player/security/GlobalSecurityConfig.java

```diff
@@ -282,7 +283,9 @@ protected void configure(HttpSecurity http) throws Exception {
             http
                     .cors()
                     .and()
-                    .addFilterBefore(restAuthenticationFilter, UsernamePasswordAuthenticationFilter.class)
+                    .httpBasic()
+                    .and()
+                    .addFilterAfter(restAuthenticationFilter, BasicAuthenticationFilter.class)
```

Comment:
Why is it necessary to have the rest auth filter now after the basic form auth filter?

Reply:
Because it throws errors when the username/password parameters are missing and stops basic auth, [but it does handle existing authentication](https://github.com/airsonic-advanced/airsonic-advanced/blob/master/airsonic-main/src/main/java/org/airsonic/player/security/RESTRequestParameterProcessingFilter.java#L86). The comment even suggest it did work at one time so I just swapped them. Order shouldn't really matter as long as it allows the next filter to attempt authentication.